File: agent/CarRentalPortal.Core/Entities/Car.cs
using CarRentalPortal.Core.Enums;
using System.Collections.Generic;

namespace CarRentalPortal.Core.Entities
{
    public class Car
    {
        public int Id { get; set; }
        public string CarBrandName { get; set; }
        public CarBrand CarBrand { get; set; }
        public string CarModelName { get; set; }
        public CarModel CarModel { get; set; }
        public int CarCategoryId { get; set; }
        public CarCategory CarCategory { get; set; }
        public int AgentId { get; set; }
        public string AgentContactInfo { get; set; }
        public short ProductionYear { get; set; }
        public FuelType FuelType { get; set; }
        public TransmissionType TransmissionType { get; set; }
        public string Color { get; set; }
        public string Location { get; set; }
        public long Mileage { get; set; }
        public long? MileageConstraint { get; set; }
        public byte NumberOfSeats { get; set; }
        public double? AverageRating { get; set; }
        public ICollection<Rental> Rentals { get; set; }
        public ICollection<Review> Reviews { get; set; }
        public ICollection<CarImage> Images { get; set; }
    }
}

File: agent/AgentUI/src/app/models/car.ts
export interface Car {
	id: number;
	agentId: number;
	agentContactInfo: string;
	carCategory: number;
	carBrand: string;
	carModel: string;
	productionYear: number;
	fuelType: FuelType;
	transmissionType: string;
	color: string;
	location: string;
	mileage: number;
	mileageConstraint?: number;
	numberOfSeats: number;
	averageRating: number;
	images: CarImage[];
	reviews: Review[];
}

export interface CarImage {
	id: number;
	uri: string;
}

export interface Review {
	id?: number;
	authorId: number;
	carId: number;
	authorDisplayName: string;
	authorEmail: string;
	rating: number;
	comment: string;
	createdOn?: Date;
}

export interface CarVm {
	cars: Car[];
}

export enum FuelType {
	Gasoline,
	Diesel,
	Hybrid,
	Electric
}

export enum TransmissionType {
	Manual,
	Automatic,
	SemiAutomatic
}